pkg/sbom/sbom.go

```diff
@@ -39,6 +31,8 @@ const (
 var ErrUnknownFormat = xerrors.New("Unknown SBOM format")
 
 func DetectFormat(r io.ReadSeeker) (Format, error) {
+	defer r.Seek(0, io.SeekStart)
```

Comment:
cool! The following code seems to be unnecessary.
https://github.com/aquasecurity/trivy/blob/9b0e9794cb94a2292c506afbff2bcd8695170558/pkg/fanal/artifact/sbom/sbom.go#L57-L60